ush/python/pygfs/task/globus_hpss.py

```diff
@@ -0,0 +1,491 @@
+#!/usr/bin/env python3
+
+import os
+import logging
+import shutil
+from time import sleep
+from typing import Any, Dict, List
+import re
+import copy
+from datetime import datetime, timezone
+
+from wxflow import AttrDict, Task, to_YMDH, logit, parse_yaml, Jinja, which, ProcessError, to_datetime
+
+logger = logging.getLogger(__name__.split('.')[-1])
+logging.basicConfig(encoding='utf-8', level=logging.DEBUG, format='%(asctime)s %(message)s')
+
+
+class GlobusHpss(Task):
+    """Task to send tarballs (created by the archive task) to HPSS via Globus
+    """
+
+    @logit(logger, name="GlobusHpss")
+    def __init__(self, config: Dict[str, Any]) -> None:
+        """Constructor for the GlobusHpss task
+
+        Parameters
+        ----------
+        config : Dict[str, Any]
+            Incoming configuration for the task from the environment
+
+        Returns
+        -------
+        None
+        """
+        super().__init__(config)
+
+        # Declare these here so the jinja-templated scripts can be shellchecked
+        cycle_YMDH = to_YMDH(self.task_config.current_cycle)
+
+        # Instantiate all of the executables we will need to run
+        self.forsven = which("forsven")
```

Comment:
```suggestion
        try:
            self.forsven = which("forsven", required=True)
        except CommandNotFoundError:
            raise CommandNotFoundError("FATAL ERROR: Could not find the forsven executable!")
```
Try try here and elsewhere where `which` is used to find a required executable.

Reply:
Thanks for the suggestion.  I've switched to that method, though I cannot test it at the moment as I cannot renew my globus connections.